lib/msal-browser/src/cache/TokenCache.ts

```diff
@@ -58,13 +58,21 @@ export class TokenCache implements ITokenCache {
             this.loadAccessToken(request, response, request.account.homeAccountId, request.account.environment, request.account.tenantId, options);
         } else if (request.authority) {
 
+            let authorityAzureCloudInstance;
+            if (request.azureAuthOptions) {
+                const tenant = request.azureAuthOptions.tenant ? request.azureAuthOptions.tenant : Constants.DEFAULT_COMMON_TENANT;
+                authorityAzureCloudInstance = `${request.azureAuthOptions.azureCloudInstance}/${tenant}/`;
+            }
+            // final authority from the request
+            const authorityUrl = authorityAzureCloudInstance ? authorityAzureCloudInstance : request.authority;
```

Comment:
Looks like this could go in common, exactly the same as Node.

Reply:
One other option is to make this a part of `authorityOptions` and change the authority url in `Authority.ts`. I added these at the client level because it is easier to make changes and more control to manipulate client specific data regarding authority in future.

Both options have their pros and cons.

Reply:
^ I think doing this in Authority would be cleaner and easier to maintain

Reply:
Moved.